src/main/java/io/stargate/sgv2/jsonapi/util/NamingValidationUtil.java

```diff
@@ -0,0 +1,23 @@
+package io.stargate.sgv2.jsonapi.util;
+
+import java.util.regex.Pattern;
+
+public interface NamingValidationUtil {
```

Comment:
ok, I see where this is going, we should look at the WhereCQLClauseAnalyzer.StatementType as a pattern. so something like...

An interface / abstract class called NamingRule with: 
- name() - friendly name 
- apply() - runs the rule on a string 
- implementations for all the naming rules 

- A static class called NamingRules that has singleton instances of each of the rule classes.